src/toil/provisioners/abstractProvisioner.py

```diff
@@ -469,26 +472,27 @@ def __init__(self):
             # Holds strings like "ssh-rsa actualKeyData" for keys to authorize (independently of cloud provider's system)
             self.sshPublicKeys = []
 
-        def addFile(self, path: str, owner: str = 'root', permissions: str = '0755', content: str = ''):
+        def addFile(self, path: str, filesystem: str = 'root', mode: int = int("0755", 8), contents: str = ''):
             """
-            Make a file on the instance with the given owner, permissions, and content.
+            Make a file on the instance with the given filesystem, mode (in decimal), and contents.
```

Comment:
The docs for the `filesystem` field is here: https://github.com/coreos/ignition/blob/spec2x/doc/configuration-v2_2.md. I don't think users can add their own filesystems so we probably don't need this.

And yeah, it would be better to do the conversion inside the function.